analyzer/codechecker_analyzer/analyzers/clangsa/analyzer.py

```diff
@@ -200,12 +202,15 @@ def construct_analyzer_cmd(self, result_handler):
                                          '-analyzer-disable-checker=' +
                                          checker_name])
 
-            # Needed for the iterator checkers.
-            analyzer_cmd.extend(['-Xclang',
-                                 '-analyzer-config',
-                                 '-Xclang',
-                                 'aggressive-binary-operation-simplification'
-                                 '=true'])
+            version_info = config.version_info
+            if version_info and version_info.major_version >= 8:
```

Comment:
Please put this to `codechecker_analyzer/analyzers/clangsa/clang_options.py` every version dependent clang analyzer options are collected there.